invenio_preservation_sync/utils.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Preservation-Sync is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Various utility functions."""
+
+from datetime import datetime
+
+import dateutil.parser
+import pytz
+import six
+from werkzeug.utils import import_string
+
+from invenio_preservation_sync.errors import MissingConfigError
+
+
+def utcnow():
+    """UTC timestamp (with timezone)."""
+    return datetime.now(tz=pytz.utc)
+
+
+def iso_utcnow():
+    """UTC ISO8601 formatted timestamp."""
+    return utcnow().isoformat()
+
+
+def parse_timestamp(x):
+    """Parse ISO8601 formatted timestamp."""
+    dt = dateutil.parser.parse(x)
+    if dt.tzinfo is None:
+        dt = dt.replace(tzinfo=pytz.utc)
+    return dt
+
+
+def obj_or_import_string(value, default=None):
+    """Import string or return object.
+
+    :params value: Import path or class object to instantiate.
+    :params default: Default object to return if the import fails.
+    :returns: The imported object.
+    """
+    if isinstance(value, six.string_types):
+        return import_string(value)
+    elif value:
+        return value
+    return default
+
+
+def get_mandatory_config(app, config_name):
```

Comment:
You could add these checks on module's initialization (`ext.py`). As part of the extension's init, some configs are mandatory. Then you can safely use them without checking for them all the time.